code/asteroid/asteroid.cpp

```diff
@@ -597,11 +597,13 @@ void asteroid_create_all()
 
 			// get a valid subtype
 			int subtype = Random::next(NUM_DEBRIS_POFS);
-			while (Asteroid_field.field_debris_type[subtype] == -1) {
-				subtype = (subtype + 1) % NUM_DEBRIS_POFS;
+			for (int i = 0; i < NUM_DEBRIS_POFS; i++) {
+				if (Asteroid_field.field_debris_type[subtype] == -1)
+					subtype = (subtype + 1) % NUM_DEBRIS_POFS;
 			}
 
-			asteroid_create(&Asteroid_field, ASTEROID_TYPE_LARGE, subtype);
+			if (subtype >= 0)
+				asteroid_create(&Asteroid_field, ASTEROID_TYPE_LARGE, subtype);
```

Comment:
Add an `else` branch with a warning.  Modders should be notified if they create an asteroid field without valid asteroids.   (And break out of the `max_asteroids` loop.)

Reply:
I put it earlier as an early return, since it's easy to explicitly catch now.